### PR TITLE
[iOS] Allow registering both legacy and raw adblock components

### PR DESCRIPTION
**Pull request by cuba**
This pull request fixes https://github.com/brave/brave-browser/issues/26951
Security ticket: https://github.com/brave/security/issues/1094

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by cuba** [comment]
@Brandon-T Any way you can help me test this one?

File: ios/browser/api/brave_shields/adblock_service.mm [hunk 1]
```diff
@@ -93,4 +93,20 @@ - (void)unregisterFilterListComponent:(AdblockFilterListCatalogEntry*)entry {
...
 
+- (void)registerComponentForComponentInfo:(ComponentRegistrationInfo*)info
+                           componentReady:(void (^)(NSString* _Nullable installPath)) componentReady {
+  brave_shields::RegisterAdBlockFiltersComponent(
+      _cus, base::SysNSStringToUTF8(info.base64PublicKey),
+      base::SysNSStringToUTF8(info.componentId),
+      base::SysNSStringToUTF8(info.title),
+      base::BindRepeating(^(const base::FilePath& install_path) {
+        const auto installPath = base::SysUTF8ToNSString(install_path.value());
+        componentReady(installPath);
```
Comment (stoletheminerals) [inline ios/browser/api/brave_shields/adblock_service.mm:104]: How do we generate `install_path` here? Do we use filter's name to generate it?

Comment (cuba) [reply ios/browser/api/brave_shields/adblock_service.mm]: @stoletheminerals  
To answer your question it came from the component updated (that part I didn't change if you look it closely).
But my approach has changed. 

I wanted to pass raw data but decided to put back the old function as it was and just pass a boolean legacy or not so the changes are much simpler now

Comment (stoletheminerals) [reply ios/browser/api/brave_shields/adblock_service.mm]: @cuba my concern was that `installPath` could be used to overwrite arbitrary files. Do you have an example for this value? If it comes from the component updater I assume it has a CRX ID for each component, but wanted to double check.

Comment (kylehickinson) [reply ios/browser/api/brave_shields/adblock_service.mm]: @stoletheminerals The install path comes from component updater: https://github.com/brave/brave-core/blob/539a4c374f3667c8d00fe9c807b396947a39a3ad/components/brave_shields/browser/ad_block_component_installer.cc#L185

**Review by kylehickinson — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

**Review by stoletheminerals — APPROVED** [review]

**Pull request merged**